operators/src/adapters/raster_overlap_adapter.rs

```diff
@@ -181,90 +209,105 @@ where
     }
 }
 
+#[allow(dead_code)]
+pub fn fold_by_blit_future<T>(
+    accu: (RasterTile2D<T>, ()),
+    tile: RasterTile2D<T>,
+) -> impl Future<Output = Result<(RasterTile2D<T>, ())>>
+where
+    T: Pixel,
+{
+    tokio::task::spawn_blocking(|| fold_by_blit_impl(accu, tile)).then(async move |x| match x {
+        Ok(r) => r,
+        Err(e) => Err(e.into()),
+    })
+}
+
+#[allow(dead_code)]
+pub fn fold_by_coordinate_lookup_future<T>(
+    accu: (RasterTile2D<T>, Vec<(GridIdx2D, Coordinate2D)>),
+    tile: RasterTile2D<T>,
+) -> impl TryFuture<Ok = (RasterTile2D<T>, Vec<(GridIdx2D, Coordinate2D)>), Error = error::Error>
+where
+    T: Pixel,
+{
+    tokio::task::spawn_blocking(|| fold_by_coordinate_lookup_impl(accu, tile)).then(
+        async move |x| match x {
+            Ok(r) => r,
+            Err(e) => Err(e.into()),
+        },
+    )
+}
+
 #[allow(dead_code)]
 #[allow(clippy::type_complexity)]
-pub fn fold_by_coordinate_lookup<T>(
-    accu: AccuType<T, Vec<(GridIdx2D, Coordinate2D)>>,
-    tile: Result<RasterTile2D<T>>,
-) -> futures::future::Ready<AccuType<T, Vec<(GridIdx2D, Coordinate2D)>>>
+#[allow(clippy::clippy::needless_pass_by_value)]
+pub fn fold_by_coordinate_lookup_impl<T>(
+    accu: (RasterTile2D<T>, Vec<(GridIdx2D, Coordinate2D)>),
+    tile: RasterTile2D<T>,
+) -> Result<(RasterTile2D<T>, Vec<(GridIdx2D, Coordinate2D)>)>
 where
     T: Pixel,
 {
-    let result: AccuType<T, Vec<(GridIdx2D, Coordinate2D)>> = match (accu, tile) {
-        (Ok((mut raster2d, lookup)), Ok(t)) => {
-            let t_union = raster2d.time.union(&t.time).unwrap();
-
-            match insert_projected_pixels(&mut raster2d, &t, lookup.iter()) {
-                Ok(_) => {
-                    raster2d.time = t_union;
-                    Ok((raster2d, lookup))
-                }
-                Err(error) => Err(error),
-            }
-        }
-        (Err(error), _) | (_, Err(error)) => Err(error),
-    };
+    let (mut accu_tile, accu_companion) = accu;
+    let t_union = accu_tile.time.union(&tile.time)?;
 
-    match result {
-        Ok(updated_raster2d) => futures::future::ok(updated_raster2d),
-        Err(error) => futures::future::err(error),
+    match insert_projected_pixels(&mut accu_tile, &tile, accu_companion.iter()) {
+        Ok(_) => {
+            accu_tile.time = t_union;
+            Ok((accu_tile, accu_companion))
+        }
+        Err(error) => Err(error),
     }
 }
 
+/// This method takes two tiles and a map from `GridIdx2D` to `Coordinate2D`. Then for all `GridIdx2D` we set the values from the corresponding coordinate in the source tile.
 pub fn insert_projected_pixels<'a, T: Pixel, I: Iterator<Item = &'a (GridIdx2D, Coordinate2D)>>(
     target: &mut RasterTile2D<T>,
     source: &RasterTile2D<T>,
     local_target_idx_source_coordinate_map: I,
 ) -> Result<()> {
+    // TODO: it would be better to run the pixel wise stuff in insert_projected_pixels in parallel...
```

Comment:
My suggestion is twofold:
1. For parallelism, instead of an iterator, insert something that is cunkable.
2. For general speedup, insert the `Vec<…>` that is behind `local_target_idx_source_coordinate_map` and return a new (smaller) version of it.

Reply:
1. iterator is chunkable
2. i  think we should do tuning as a new task

Reply:
Okay, maybe you can create one with some details on what to improve.